crates/gpui/src/platform/linux/wayland/client.rs

```diff
@@ -63,8 +65,13 @@ pub(crate) struct WaylandClientStateInner {
     loop_handle: Rc<LoopHandle<'static, ()>>,
 }
 
+pub(crate) struct CursorStateInner {
+    cursor_icon_name: String,
+    cursor: Option<Cursor>,
+}
+
 #[derive(Clone)]
-pub(crate) struct WaylandClientState(Rc<RefCell<WaylandClientStateInner>>);
+pub(crate) struct WaylandClientState(Rc<RefCell<WaylandClientStateInner>>, Rc<RefCell<CursorStateInner>>);
```

Comment:
I wanted to track the cursor state as part of the wayland client state inner. But I couldn't borrow it in `set_cursor_style()`, because it complained "already borrowed". So I went with tracking the cursor state separately.